model/service_test.go

```diff
@@ -90,6 +90,55 @@ func TestServiceTransform(t *testing.T) {
 				},
 			},
 		},
+		{
+			Service: Service{
+				Name: serviceName,
+				Node: ServiceNode{Name: serviceNodeName},
+				Target: &ServiceTarget{
+					Name: "",
+					Type: "test",
+				},
+			},
+			Fields: mapstr.M{
+				"name": "myService",
+				"node": mapstr.M{"name": serviceNodeName},
+				"target": mapstr.M{
+					"type": "test",
+				},
+			},
+		},
+		{
+			Service: Service{
+				Name: serviceName,
+				Node: ServiceNode{Name: serviceNodeName},
+				Target: &ServiceTarget{
+					Name: "test",
+					Type: "",
+				},
+			},
+			Fields: mapstr.M{
+				"name": "myService",
+				"node": mapstr.M{"name": serviceNodeName},
+				"target": mapstr.M{
+					"name": "test",
+					"type": "",
+				},
+			},
+		},
+		{
+			Service: Service{
+				Name: serviceName,
+				Node: ServiceNode{Name: serviceNodeName},
+				Target: &ServiceTarget{
+					Name: "",
+					Type: "",
+				},
```

Comment:
Can this really happen? If it can't, maybe we should tighten up the JSON-Schema so constraints so if target is specified, either field must be present.

Reply:
Hmm, you are correct. There will be 2 cases:

1. Older agents will not be sending us target information so target itself will be `nil` however it will be inferred by the resource field => as per the spec this is where the `target.type` can be nil but both `name` and `type` shouldn't be `nil`.
2.  For newer agents, as per the spec, `service.target.type` should always be present, if it is not present then even name is ignored.

Based on this I think we can add a required constraint on `target.type` if target is provided.

Reply:
That sounds good to me. I don't know if it needs to be captured in the spec or not, but can you please run it by the agent devs to make sure they're at least aware?

Reply:
Updated the code to validate empty target (without both `type` or `name` provided)